python/ray/util/spark/cluster_init.py

```diff
@@ -107,11 +107,14 @@ def wait_until_ready(self):
             raise RuntimeError(
                 "The ray cluster has been shut down or it failed to start."
             )
+
         try:
-            # connect to the ray cluster.
-            ray_ctx = ray.init(address=self.address)
-            webui_url = ray_ctx.address_info.get("webui_url", None)
-            if webui_url:
+            ray.init(address=self.address)
+
+            if (
+                self.ray_dashboard_port is not None and
+                check_port_open(self.address.split(":")[0], self.ray_dashboard_port)
```

Comment:
If port is open, doesn't this mean the process is not started?

Reply:
`check_port_open` returns True if the port is bound to a TCP service and service is up.

The function name may be not good, what about rename it to "is_port_in_use" ?

Reply:
Yeah that sounds good.